### PR TITLE
Create labels when volume exists only remotely

### PR DESCRIPTION
**Pull request by cpuguy83**
Before this, if a volume exists in a driver but not in the local cache,
the store would just return a bare volume. This means that if a user
supplied options or labels, they will not get stored.

Instead only return early if we have the volume stored locally. Note
this could still have an issue with labels/opts passed in by the user
differing from what is stored, however this isn't really a new problem.

~~There is a minor behavior change in the interaction with the volume
driver in that `create` is called each time instead of doing a
get and then create. This shouldn't be a problem as the get was just a
best effort and create is expected to be idempotent.~~

This fixes a problem where if there is a shared storage backend between
two docker nodes, a create on one node will have labels stored and a
create on the other node will not.

### CODE REVIEW

**Comment by cpuguy83** [comment]
ping @srust @clintkitson @gourao

**Comment by sixcounts** [comment]
@tiborvass any idea when this pull request will get merged for an engine release?

Thank you

**Comment by sixcounts** [comment]
@tiborvass any ETA as to when this will get merged into which release? I have a customer that has been having problems with this Docker Certified Rex-Ray PlugIn for EMC's Isilon for several months now.

Thank you,

Josh

**Comment by sixcounts** [comment]
@cpuguy83 @tiborvass there is a meeting with this customer this week and we need a solid update to give them. Can someone provide an update on to when this will be merged and what release will address it? This stemmed from the Docker Certified Plugin for Rex-Rex for EMC Isilon:

https://github.com/codedellemc/rexray/issues/989

Thank you,

Josh

**Comment by thaJeztah** [comment]
@sixcounts for certified plugins it'll have to end up in a Docker EE release, so that would have to be back-ported by Docker; it's best to open a backport request for that internally.

**Comment by sixcounts** [comment]
@thaJeztah I have opened internal escalation:

https://github.com/docker/escalation/issues/209

for this issue.

File: volume/store/store.go [hunk 1]
```diff
@@ -385,32 +385,33 @@ func (s *VolumeStore) create(name, driverName string, opts, labels map[string]st
...
 
 	vd, err := volumedrivers.CreateDriver(driverName)
-
 	if err != nil {
 		return nil, &OpErr{Op: "create", Name: name, Err: err}
 	}
 
 	logrus.Debugf("Registering new volume reference: driver %q, name %q", vd.Name(), name)
 
-	if v, _ := vd.Get(name); v != nil {
```
Comment (cpuguy83) [inline volume/store/store.go]: Maybe I don't need to get rid of this... may need to revisit, but just need to ensure that labels are stored locally.

**Comment by vieux** [comment]
@cpuguy83 if is ready to merge ?

**Comment by cpuguy83** [comment]
I'm happy with it, no behavior change in volume driver interaction like was originally implemented. 

On mobile so I can't inspect that windows error at the moment.

**Comment by vieux** [comment]
@cpuguy83 

```
21:33:58 ----------------------------------------------------------------------
21:33:58 FAIL: docker_cli_logs_test.go:92: DockerSuite.TestLogsTail
21:33:58 
21:33:58 docker_cli_logs_test.go:113:
21:33:58     c.Assert(lines, checker.HasLen, testLen+1)
21:33:58 ... obtained []string = []string{""}
21:33:58 ... n int = 101
21:33:58 
21:34:05 
21:34:05 ----------------------------------------------------------------------
```

I restarted

**Comment by vieux** [comment]
LGTM

**Comment by tonistiigi** [comment]
LGTM

**Pull request merged**